### DISCUSSION TITLE
How to start draw by custom trigger?

### DISCUSSION THREAD

**Discussion by vitonsky** [fabricjs/fabric.js#7948] [Q&A]
By default we start draw by press mouse or by start touch one finger. I need to start draw by touch move, how i can do it?

**Comment by ShaMan123** [comment]
create a custom brush or subclass an existing one.

**Reply by vitonsky** [reply]
How i can control the drawing?

For example i will introduce subclass of Pencil. How to trigger start/stop drawing?

**Reply by ShaMan123** [reply]
Look at the source code https://github.com/fabricjs/fabric.js/tree/master/src/brushes

**Reply by vitonsky** [reply]
For example i want to use https://github.com/fabricjs/fabric.js/blob/master/src/brushes/pencil_brush.class.js

How to trigger drawing? I should not to call an `onMouseDown`, but should execute all code inside this method?

**Comment by ShaMan123** [comment]
You are a dev. Figure it out 🙃
Search for onMouseDownInDrawingMode.

בתאריך יום ד׳, 18 במאי 2022, 12:43, מאת vitonsky ***@***.***

**Reply by vitonsky** [reply]
I'm developer, but not a wizard, i rely on documentation.

Could you improve documentation about use brushes? It will useful for all users of fabric

**Reply by vitonsky** [reply]
We should add information about life cycle of brush. How to start draw, how to stop, which methods call and which minimal state changes is required

**Reply by ShaMan123** [reply]
> I'm developer, but not a wizard, i rely on documentation.

Yes, you have a point.
My best advice is to reverse engineer by subclassing PencilBrush and planting `console.log(new Error('I will give the stack trace'))` in methods of interest.

**Reply by ShaMan123** [reply]
> We should add information about life cycle of brush. How to start draw, how to stop, which methods call and which minimal state changes is required

BaseBrush is only thing near a lifecycle and that's hardly close enough
